src/toolchain/export-web.md

```diff
@@ -0,0 +1,155 @@
+<!--
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+
+# Intro to Web Builds
+
+Web builds are a fair bit more difficult to get started with compared to native builds.
+This will be a complete guide on how to get things compiled.
+However, setting up a web server to host and share your game is considered out of scope of this guide, and is best explained elsewhere.
+
+Information to consider before starting:
+
+```admonish warning
+Web support with `gdext` is experimental and should be understood as such before proceeding.
+```
+
+---
+
+
+## Installing Pre-Requisites
+
+Install a nightly build of `rustc`, the `wasm32-unknown-emscripten` target for `rustc`, and `rust-src`.
+The reason that a nightly `rustc` is required is that nightly is required to build `std` (`-Zbuild-std`).
+Assuming that Rust was installed with `rustup`, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+Next, install Emscripten.  The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+```sh
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install 3.1.39
+./emsdk activate 3.1.39
+source ./emsdk.sh     (or ./emsdk.bat on windows)
+```
+
+It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+This is platform-specific.
+
+---
+
+
+## Project Configuration
+
+Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
+
+```toml
+godot = { git = "https://github.com/godot-rust/gdext", branch = "master", features = ["experimental-wasm", "lazy-function-tables"] }
```

Comment:
Since this line gets very long, we should probably format the TOML like this instead:
```suggestion
[dependencies.godot]
git = "https://github.com/godot-rust/gdext"
branch = "master"
features = ["experimental-wasm", "lazy-function-tables"]
```

Also, do you recommend using `lazy-function-tables` as a default? If yes, should we later consider enabling this implicitly if `experimental-wasm` is chosen?

In other words, can you build WASM without `lazy-functino-tables`, and what's the cost?

Reply:
Without `lazy-function-tables` it takes about 10 minutes to compile on release, vs about 30 seconds with it, on my m3 pro mac.  Also, with it off it seems to crash whilst generating the cache table, for me at least.  I hadn't spent that much time looking into it, but I guess it's more of a "very strongly recommended" instead of required? Hopefully...

Reply:
I see, thanks for elaboration!

In that case I would add a comment to the above TOML snippet, saying that the feature is not strictly required, but can help compile times (at the cost of a tiny bit slower function calls).

Reply:
^